modules/generic-scala3/src/main/scala/pureconfig/generic/scala3/HintsAwareConfigConvertDerivation.scala

```diff
@@ -0,0 +1,11 @@
+package pureconfig
+package generic
+package scala3
+
+import scala.deriving.Mirror
+
+trait HintsAwareConfigConvertDerivation extends HintsAwareConfigWriterDerivation, HintsAwareConfigReaderDerivation:
+  inline def deriveConvert[A: Mirror.Of]: ConfigConvert[A] =
```

Comment:
Is there a need for a separate trait and object for deriving `ConfigConvert`? Through the rest of the codebase we always kept the concept of `ConfigConvert` very lightweight - we always try to define it as a function of a `ConfigReader` + `ConfigWriter` rather than giving it its own logic. What do you think of putting this definition of `deriveConvert` directly inside `semiauto` to simplify things?

Reply:
I intentionally left the `semiauto` as simple as possible to treat it as place for exports, but sure, I can do that.

Reply:
Yeah, I understand, it's mostly a question of taste 🙂 I consider those cases different as `deriveConvert` (and creation of `ConfigConvert` instances in general) is supposed to be just a convenience method, equivalent to providing `given` instances for `ConfigReader` and `ConfigWriter`. Hiding that away into a full fledged trait equivalent to the other two could open a path to it being more than that.